### PR TITLE
cross build setting for `javax` and `jakarta`

### PR DESCRIPTION
**Pull request by xuwei-k**
- https://github.com/sbt/sbt-projectmatrix
- http://eed3si9n.com/parallel-cross-building-using-sbt-projectmatrix/

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -67,48 +70,63 @@ lazy val scalatraSettings = Seq(
...
+
+val versionJavax = Def.settings(
+  version := {
+    val v = version.value
+    val snapshotSuffix = "-SNAPSHOT"
+    val suffix = "-javax"
+    if (v.endsWith(snapshotSuffix)) {
+      v.dropRight(snapshotSuffix.length) + suffix + snapshotSuffix
+    } else {
+      v + suffix
```
[the remark this answers is not in the document]
Comment (xuwei-k) [reply build.sbt:87]: current setting (add suffix both)

```
org/scalatra/scalatra-auth_2.12/3.0.0-M5-jakarta/scalatra-auth_2.12-3.0.0-M5-jakarta.jar
org/scalatra/scalatra-auth_2.12/3.0.0-M5-javax/scalatra-auth_2.12-3.0.0-M5-javax.jar
org/scalatra/scalatra-auth_2.13/3.0.0-M5-jakarta/scalatra-auth_2.13-3.0.0-M5-jakarta.jar
org/scalatra/scalatra-auth_2.13/3.0.0-M5-javax/scalatra-auth_2.13-3.0.0-M5-javax.jar
org/scalatra/scalatra-auth_3/3.0.0-M5-jakarta/scalatra-auth_3-3.0.0-M5-jakarta.jar
org/scalatra/scalatra-auth_3/3.0.0-M5-javax/scalatra-auth_3-3.0.0-M5-javax.jar
org/scalatra/scalatra-cache_2.12/3.0.0-M5-jakarta/scalatra-cache_2.12-3.0.0-M5-jakarta.jar
org/scalatra/scalatra-cache_2.12/3.0.0-M5-javax/scalatra-cache_2.12-3.0.0-M5-javax.jar
org/scalatra/scalatra-cache_2.13/3.0.0-M5-jakarta/scalatra-cache_2.13-3.0.0-M5-jakarta.jar
org/scalatra/scalatra-cache_2.13/3.0.0-M5-javax/scalatra-cache_2.13-3.0.0-M5-javax.jar
org/scalatra/scalatra-cache_3/3.0.0-M5-jakarta/scalatra-cache_3-3.0.0-M5-jakarta.jar
org/scalatra/scalatra-cache_3/3.0.0-M5-javax/scalatra-cache_3-3.0.0-M5-javax.jar
...
```

[the remark this answers is not in the document]
Comment (takezoe) [reply build.sbt]: The current setting looks good!

**Comment by dalbani** [comment]
Thanks @xuwei-k for your work.
I'd like to test `M5` but I don't see it on Maven Central.
Could you publish the artifacts?

**Comment by takezoe** [comment]
@dalbani M4 with jakarta suffix is available.

**Comment by dalbani** [comment]
Yes, sure, I've tested my app (successfully) against M4, but, just to be sure, I'd like to test if the `org.scalatra.ServletCompat` construct works fine as well.

**Comment by dalbani** [comment]
I come back to the question I mentioned in my previous message.
I'd very much like to test integrating the current code from `main` in my application.
I thought there would an `M5` being released, but now I see that it's back to `SNAPSHOT` now in:
https://github.com/scalatra/scalatra/blob/b9f509c3d3854bd05e77ead23b0787713862ce93/version.sbt#L1

Could you clarify?

**Comment by takezoe** [comment]
I guess that was not very intentional in this pull request. Let's update the version to M5 and release it. Sorry for my late response.

**Pull request merged**